config/management/src/validator_config.rs

```diff
@@ -53,6 +70,11 @@ impl ValidatorConfig {
             .append_prod_protos(validator_network_key, HANDSHAKE_VERSION);
         let raw_validator_address = RawNetworkAddress::try_from(&validator_address)
             .map_err(|e| Error::UnexpectedError(format!("(raw_validator_address) {}", e)))?;
+        let enc_validator_address = raw_validator_address
+            .encrypt(&root_key, key_version, &sender, sequence_number, addr_idx)
+            .map_err(|e| Error::UnexpectedError(format!("(enc_validator_address) {}", e)))?;
+        let raw_enc_validator_address = RawEncNetworkAddress::try_from(&enc_validator_address)
```

Comment:
Should we come up with a new name for these?

I found it confusing earlier with the:
* NetworkAddress (Uses Protocols)
* RawNetworkAddress (Uses encoded protocols)

And we've now added:
* RawEncNetworkAddress (Uses Encrypted, encoded protocols) but, for me looking at it with no knowledge might think it's just encoded.

It might not be for right now, but we should think of a way to simplify it or rename it, I don't have a good suggestion off the top of my head right now.

Reply:
hmm, maybe "Raw" is the wrong prefix? would it be less confusing if it was like `SerNetworkAddress` (i.e., serialized network address)? personally, i think `EncryptedNetworkAddress` is getting a bit too long : ( maybe we shorten `NetworkAddress` to `NetAddr`?

one of the reasons for the thin `RawNetworkAddress`/`RawEncNetworkAddress` wrappers is to ensure we can't accidentally deserialize as the wrong thing.

for example, the on-chain representation of a `validator_network_address` in each `ValidatorConfig` is just a `vector<u8>` in Move, but we represent it as a `RawEncNetworkAddress` in rust so it's not possible (or at least harder) to accidentally deserialize it as something that's not an `EncNetworkAddress`.

in my mind the types sort of encode the transformations

NetworkAddress <--> RawNetworkAddress <--> EncNetworkAddress <--> RawEncNetworkAddress

Reply:
Will we always know ahead of time if it's encrypted or not?

Maybe `Bin` instead of `Raw` to point out it's binary, but I don't think that part is really important.  

I guess my nit here is EncNetworkAddress isn't really an Encrypted NetworkAddress, but instead an encrypted RawNetworkAddress.

Reply:
> Will we always know ahead of time if it's encrypted or not?

yes, the `validator_network_address` field will only contain serialized encrypted network addresses (hoping to have time to rename these on-chain things a bit : p)

> I guess my nit here is EncNetworkAddress isn't really an Encrypted NetworkAddress, but instead an encrypted RawNetworkAddress.

I thought about this as well, but "RawEncRawNetworkAddress" looks pretty wild. it's definitely a more consistent naming scheme though

Reply:
Let's just revisit the naming for those after this then